crates/iota-core/src/jsonrpc_index.rs

```diff
@@ -1324,41 +1319,27 @@ impl IndexStore {
     /// the `all_balance` cache. Only on the second cache miss, we go to the
     /// database (expensive) and update the cache. Notice that db read is
     /// done with `spawn_blocking` as that is expected to block
-    pub async fn get_balance(
-        &self,
-        owner: IotaAddress,
-        coin_type: TypeTag,
-    ) -> IotaResult<TotalBalance> {
+    pub fn get_balance(&self, owner: IotaAddress, coin_type: TypeTag) -> IotaResult<TotalBalance> {
         let force_disable_cache = read_size_from_env(ENV_VAR_DISABLE_INDEX_CACHE).unwrap_or(0) > 0;
         let cloned_coin_type = coin_type.clone();
         let metrics_cloned = self.metrics.clone();
         let coin_index_cloned = self.tables.coin_index.clone();
         if force_disable_cache {
-            return spawn_blocking(move || {
-                Self::get_balance_from_db(
-                    metrics_cloned,
-                    coin_index_cloned,
-                    owner,
-                    cloned_coin_type,
-                )
-            })
-            .await
-            .unwrap()
-            .map_err(|e| IotaError::Execution(format!("Failed to read balance frm DB: {e:?}")));
+            Self::get_balance_from_db(metrics_cloned, coin_index_cloned, owner, cloned_coin_type)
```

Comment:
I think there should be return instead of `?`.

Reply:
good catch, I also think so